src/Simplex/Chat/Library/Commands.hs

```diff
@@ -3045,37 +3039,76 @@ processChatCommand' vr = \case
             pure (gId, chatSettings)
         _ -> throwChatError $ CECommandError "not supported"
       processChatCommand $ APISetChatSettings (ChatRef cType chatId) $ updateSettings chatSettings
-    connectPlan :: User -> AConnectionLink -> CM (AConnectionRequestUri, ConnectionPlan)
-    connectPlan user (ACL SCMInvitation cLink) = do
-      cReq <- getLinkConnReq user cLink
-      fmap (ACR SCMInvitation cReq,) $ withFastStore' (\db -> getConnectionEntityByConnReq db vr user $ cReqSchemas cReq) >>= \case
+    connectPlan :: User -> AConnectionLink -> CM (ACreatedConnLink, ConnectionPlan)
+    connectPlan user (ACL SCMInvitation cLink) = case cLink of
+      CLFull cReq -> invitationReqAndPlan cReq Nothing
+      CLShort sLnk ->
+        withFastStore' (\db -> getConnectionEntityViaShortLink db vr user sLnk) >>= \case
+          Just (cReq, ent) ->
+            (ACCL SCMInvitation (CCLink cReq (Just sLnk)),) <$> (invitationEntityPlan ent `catchChatError` (pure . CPError))
+          Nothing -> getShortLinkConnReq user sLnk >>= (`invitationReqAndPlan` Just sLnk)
+      where
+        invitationReqAndPlan cReq sLnk_ = do
+          plan <- inviationRequestPlan user cReq `catchChatError` (pure . CPError)
+          pure (ACCL SCMInvitation (CCLink cReq sLnk_), plan)
+    connectPlan user (ACL SCMContact cLink) = case cLink of
+      CLFull cReq -> contactReqAndPlan cReq Nothing
+      CLShort sLnk@(CSLContact _ ct _ _) -> case ct of
+        CCTContact ->
+          withFastStore' (\db -> getUserContactLinkViaShortLink db user sLnk) >>= \case
+            Just (UserContactLink (CCLink cReq _) _) -> pure (ACCL SCMContact $ CCLink cReq (Just sLnk), CPContactAddress CAPOwnLink)
+            Nothing -> getShortLinkConnReq user sLnk >>= (`contactReqAndPlan` Just sLnk)
+        CCTGroup ->
+          withFastStore' (\db -> getGroupInfoViaUserShortLink db vr user sLnk) >>= \case
+            Just (cReq, g) -> pure (ACCL SCMContact $ CCLink cReq (Just sLnk), CPGroupLink (GLPOwnLink g))
+            Nothing -> getShortLinkConnReq user sLnk >>= (`contactReqAndPlan` Just sLnk)
```

Comment:
could be smell that it first differentiates contact/group here, then both do contactReqAndPlan, which in turn again differentiates contact/group but differently.

shouldn't it be streamlined to all contact related separately, all group related separately?

contactReqAndPlan is used elsewhere too, so maybe it should be split into "contactReqAndPlanAddr", "contactReqAndPlanGroup", these two reused here and united contactReqAndPlan used where it's needed?